tasks/build.py

```diff
@@ -1,30 +1,113 @@
+                log("  PR file '%s' white listed" % (pr_file.filename))
+                easystack = repo.get_contents(pr_file.filename, ref=pr.head.ref)
+
+                # create directory for the "file" (including directory) from PR
+                stack_job_dir = os.path.join(job_run_dir, stack_version)
+                if not os.path.exists(stack_job_dir):
+                    os.makedirs(stack_job_dir)
+
+                for arch_target,slurm_opt in arch_target_map.items():
+                    arch_job_dir = os.path.join(stack_job_dir, arch_target.replace("/","_"))
+                    if not os.path.exists(arch_job_dir):
+                        os.makedirs(arch_job_dir)
+
+                    with open(os.path.join(arch_job_dir, stack_file), 'w') as easystack_file:
+                        easystack_file.write(easystack.decoded_content.decode('UTF-8'))
+
+                    # enlist jobs to proceed
+                    jobs.append([stack_version,arch_target,slurm_opt,arch_job_dir,stack_file])
+
+    log("  %d jobs to proceed after applying white list" % len(jobs))
+    if len(jobs) > 0:
+        log(json.dumps(jobs, indent=4))
 
-    # Run the build job submission script
+    # Run jobs the build job submission script
     # Submit functionality should probably moved here at some point, now all part of the Bash script
-    submit_cmd = subprocess.run(['./automatic-build.sh', jobdir], cwd=SCRIPTS_DIR, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    log("Submit command executed!\nStdout: %s\nStderr: %s" % (submit_cmd.stdout, submit_cmd.stderr))
+    for job in jobs:
+        log("  Create stack given by '%s'\n    for version '%s'\n    on target '%s'\n    with options '%s'\n    in directory '%s'" % (job[4],job[0],job[1],job[2],job[3]))
+        # instead of setting these env variables maybe store these
+        # in some '_env' file in the job directory, this (writing
+        # into the file '_env') is anyhow done by the batch script
+        # and later on '_env' is read by the modified EESSI install
+        # script
+        d = dict(os.environ)
+        d["SOFTWARE_FILE"] = job[4]
+        d["SCRIPTS_DIR"] = scripts_dir
+        d["LOCAL_TMP"] = local_tmp
+        d["BUILD_TARGET_ARCHITECTURE"] = job[1]
+        d["BUILD_EESSI_VERSION"] = job[0]
+        log("Submit job with '%s %s %s/%s' from directory '%s'" % (submit_command, job[2], scripts_dir, build_job_script, job[3]))
+        # alternative to command_line as string would be a list of strings each without a whitespace, e.g.,
+        # [submit_command, job[2] /*splitted*/, scripts_dir + '/' + build_job_script]
+        command_line = submit_command + ' ' + job[2] + ' ' + scripts_dir + '/' + build_job_script;
+        submitted_job = subprocess.run(command_line, shell=True, cwd=job[3], env=d, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        log("Submit command executed!\nStdout: %s\nStderr: %s" % (submitted_job.stdout, submitted_job.stderr))
```

Comment:
Future work: also check exit code, so we can deal with problems with job submission, etc.

Reply:
Created issue #6 as a reminder for this.